patches/0083-lavfi-dnn-Add-OpenVINO-API-2.0-support.patch

```diff
@@ -0,0 +1,1224 @@
++{
++    int ret;
++    ov_dimension_t dims[4] = {{1, 1}, {1, 1}, {input_height, input_height}, {input_width, input_width}};
++    OV2Model *ov2_model = model;
++    OV2Context *ctx = &ov2_model->ctx;
++    TaskItem task;
++    OV2RequestItem *request;
++    ov_status_e status;
++    ov_shape_t input_shape = {0};
++    ov_partial_shape_t partial_shape;
++    DNNExecBaseParams exec_params = {
++        .input_name   = input_name,
++        .output_names = &output_name,
++        .nb_output    = 1,
++        .in_frame     = NULL,
++        .out_frame    = NULL,
++    };
++
++    if (ov2_model->model->func_type != DFT_PROCESS_FRAME) {
++        av_log(NULL, AV_LOG_ERROR, "Get output dim only when processing frame.\n");
++        return AVERROR(EINVAL);
++    }
++
++    if (ctx->options.input_resizable) {
++        if (!ov_model_is_dynamic(ov2_model->ov_model)) {
++            status = ov_partial_shape_create(4, dims, &partial_shape);
++            status = ov_const_port_get_shape(ov2_model->input_port, &input_shape);
++            input_shape.dims[2] = input_height;
++            input_shape.dims[3] = input_width;
++            if (status != OK) {
++                av_log(ctx, AV_LOG_ERROR, "Failed create shape for model input resize.\n");
++                goto err;
++            }
++
++            status = ov_shape_to_partial_shape(input_shape, &partial_shape);
++            if (status != OK) {
++                av_log(ctx, AV_LOG_ERROR, "Failed create partial shape for model input resize.\n");
++                goto err;
++            }
++
++            status = ov_model_reshape_single_input(ov2_model->ov_model, partial_shape);
++            if (status != OK) {
++                av_log(ctx, AV_LOG_ERROR, "Failed to reszie model input.\n");
++                goto err;
++            }
++        } else {
++            avpriv_report_missing_feature(ctx, "Do not support dynamic model.");
++            goto err;
++        }
++    }
++
++    status = ov_model_const_output_by_index(ov2_model->ov_model, 0, &ov2_model->output_port);
++    if (status != OK) {
++        av_log(ctx, AV_LOG_ERROR, "Failed to get output port.\n");
++        goto err;
++    }
++
++    init_model_ov2(ov2_model, input_name, output_name);
++
++    ret = ff_dnn_fill_gettingoutput_task(&task, &exec_params, ov2_model, input_height, input_width, ctx);
++    if (ret != 0) {
++        goto err;
++    }
++
++    ret = extract_lltask_from_task(ov2_model->model->func_type, &task, ov2_model->lltask_queue, NULL);
++    if (ret != 0) {
++        av_log(ctx, AV_LOG_ERROR, "unable to extract inference from task.\n");
++        goto err;
++    }
++
++    request = ff_safe_queue_pop_front(ov2_model->request_queue);
++    if (!request) {
++        av_log(ctx, AV_LOG_ERROR, "unable to get infer request.\n");
++        ret = AVERROR(EINVAL);
++        goto err;
++    }
++
++    ret = execute_model_ov2(request, ov2_model->lltask_queue);
```

Comment:
The return value from execute_model_ov2 is a dnn error code, however the expected value from get_output_ov2 is AVERROR.

Reply:
Yes, this is an issue. I use a new func `ov2_map_error` to map OpenVINO error code to FFmpeg av_error, just like qsv did.